packages/preferences/src/browser/UI/settings-ui-plus-array.tsx

```diff
@@ -0,0 +1,90 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as React from 'react';
+
+const { useState } = React;
+
+interface SettingsUIPlusArrayProps {
+    values: string[];
+    preferenceID: string;
+    removeValue(id: string, i: number): void;
+    submit(id: string, value: string): void;
+}
+
+export function SettingsUIPlusArray(
```

Comment:
Maybe I'm lazy, but could you add a docstring just walking us through what this component is meant to help with? I will keep reading the implementation to get an idea, but I think it would be nice to document these things more :)

Reply:
Added a docstring. Is it up to par?